include/deal.II/multigrid/mg_transfer.h

```diff
@@ -143,6 +147,25 @@ namespace internal
   };
 
 #endif
+
+#ifdef DEAL_II_WITH_PETSC
+  template <>
+  struct MatrixSelector<dealii::PETScWrappers::MPI::Vector>
+  {
+    typedef ::dealii::DynamicSparsityPattern Sparsity;
+    typedef ::dealii::PETScWrappers::MPI::SparseMatrix Matrix;
+
+    template <typename SparsityPatternType, typename DoFHandlerType>
+    static void reinit(Matrix &matrix, Sparsity &, int level, const SparsityPatternType &sp, const DoFHandlerType &dh)
+    {
+      // Reinit PETSc matrix
+      matrix.reinit(dh.locally_owned_mg_dofs(level+1),
+                    dh.locally_owned_mg_dofs(level),
+                    sp, communicator);
```

Comment:
You are missing to declare `communicator` by something like (copied from above)
```
parallel::Triangulation<DoFHandlerType::dimension,DoFHandlerType::space_dimension> *dist_tria =
    dynamic_cast<const parallel::Triangulation<DoFHandlerType::dimension,DoFHandlerType::space_dimension>*>
         (&(dh.get_triangulation()));
MPI_Comm communicator = dist_tria != nullptr ? dist_tria->get_communicator() : MPI_COMM_SELF;
```

Reply:
Whoops. Will fix that when I get back to work tomorrow. Sorry for pushing that broken commit, I tend to commit a lot in-between changes and @drwells already mentioned that the commits need to be re-organized in the end so I packed everything into the same branch.